### DISCUSSION TITLE
Help with Deleting Duplicates Using immich-go When Filenames Differ Slightly

### DISCUSSION THREAD

**Discussion by Fenribel** [simulot/immich-go#445] [Q&A]
Hello,

I’m trying to use immich-go to delete duplicate photos from my Immich server. The tool looks promising, but I noticed that it deletes files based on identical filenames. In my case, I have duplicates where the filenames differ slightly, such as "filename.jpg" and "filename_compressed.jpg."

Is there a way to configure immich-go to specifically delete files with "_compressed" in their name if there is a corresponding file without this suffix (e.g., delete "filename_compressed.jpg" if "filename.jpg" exists)?

Thank you for your help!

**Comment by simulot** [comment]
Immich-go is meant to remove duplicates resulting of the google photo storage optimization. The version coming from the takeout is discarded when the original having the same name and the same date of capture is found.

So at the moment, it's not doable.

**Reply by Fenribel** [reply]
Thank you for your response! I just want to make sure I understand the duplicate function in immich-go correctly (https://github.com/simulot/immich-go?tab=readme-ov-file#command-duplicate). As I understand it, the function does not rely on the origin of the photos—it simply removes duplicates based on filenames and metadata?

My main question is whether I can customize this function further to remove not only files with identical names but also those with slightly different names (for example, when a duplicate has a "_compressed" suffix). If this feature is not currently implemented, I would be willing to try contributing to your program to add it. I don’t have much experience in programming, so I can’t guarantee success, but I’m willing to give it a shot.
